fda/functional_data.py

```diff
@@ -516,21 +522,276 @@ def shift(self, shifts, *, restrict_domain=False, extrapolation=None,
-                Defaults to matplotlib current axis.
-            **kwargs: keyword arguments to be passed to the
-                matplotlib.pyplot.plot function.
+            nrows(int, optional): designates the number of rows of the figure to
+                plot the different dimensions of the image. ncols must be also
+                be customized in the same call.
+            ncols(int, optional): designates the number of columns of the figure
+                to plot the different dimensions of the image. nrows must be
+                also be customized in the same call.
 
         Returns:
-            List of lines that were added to the plot.
+            fig (figure object): figure object initialiazed.
+            ax (axes object): axes of the initialized figure.
 
         """
-        pass
+
+        if self.ndim_domain == 1:
+            projection = None
+        else:
+            projection = '3d'
+
+        if ncols is None and nrows is None:
+            ncols = int(numpy.ceil(numpy.sqrt(self.ndim_image)))
+            nrows = int(numpy.ceil(self.ndim_image / ncols))
+        elif ncols is None and nrows is not None:
+            nrows = int(numpy.ceil(self.ndim_image / nrows))
+        elif ncols is not None and nrows is None:
+            nrows = int(numpy.ceil(self.ndim_image / ncols))
+
+        fig = plt.gcf()
+        axes = fig.get_axes()
+
+        # If it is not empty
+        if len(axes) != 0:
+            # Gets geometry of current fig
+            geometry = (fig.axes[0]
+                        .get_subplotspec()
+                        .get_topmost_subplotspec()
+                        .get_gridspec().get_geometry())
+
+            # Check if the projection of the axes is the same
+            if projection == '3d':
+                same_projection = all(a.name == '3d' for a in axes)
+            else:
+                same_projection = all(a.name == 'rectilinear' for a in axes)
+
+            # If compatible uses the same figure
+            if (same_projection and geometry == (nrows, ncols) and
+                len(axes) == self.ndim_image):
+                 return fig, axes
+
+            else: # Create new figure if it is not compatible
+                fig = plt.figure()
+
+
+        for i in range(self.ndim_image):
+            fig.add_subplot(nrows, ncols, i + 1, projection=projection)
+
+        if ncols > 1 and self.axes_labels is not None and self.ndim_image > 1:
+            plt.subplots_adjust(wspace=0.4)
+
+        if nrows > 1 and self.axes_labels is not None and self.ndim_image > 1:
+            plt.subplots_adjust(hspace=0.4)
+
+        ax = fig.get_axes()
+
+        return fig, ax
+
+    def set_labels(self, fig=None, ax=None):
+        """Set labels if any.
+
+        Args:
+            fig (figure object): figure object containing the axes that
+                implement set_xlabel and set_ylabel, and set_zlabel in case
+                of a 3d projection.
+
+        """
```

Comment:
This piece of code I think it can be changed to:
¨¨¨
if self.dataset_label is not None:
   fig.suptitle(self.dataset_label)
¨¨¨
I had to do this checks because I did not want to put the same title to all the subplots of the same figure in case the `fig` parameter was None, but now is not going to be None due to the modification in the returning objects of `generic_potting_checks`.